src/formatDate.js

```diff
@@ -1,14 +1,54 @@
 'use strict';
 
-/**
- * @param {string} date
- * @param {string[]} fromFormat
- * @param {string[]} toFormat
- *
- * @returns {string}
- */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const prevSplitter = fromFormat[3];
```

Comment:
The code assumes that the separator is always at the third index of the format string. This might not be true for all formats. Consider using a more reliable method to determine the separator.